compiler/ballerina-lang/src/main/java/io/ballerina/compiler/api/impl/BallerinaSemanticModel.java

```diff
@@ -294,7 +297,25 @@ private BSymbol findSymbolAtCursorPosition(Document sourceDocument, LinePosition
             return Optional.empty();
         }
 
-        return Optional.ofNullable(typesFactory.getTypeDescriptor(node.getDeterminedType()));
+        // The determined type of invocation with a start action is set to be future. Therefore, the constraint type
+        // is obtained if the given range is pointed to the invocation function.
+        BType determinedType = getDeterminedType(node, range);
+
+        return Optional.ofNullable(typesFactory.getTypeDescriptor(determinedType));
+    }
+
+    private BType getDeterminedType(BLangNode node, LineRange range) {
+        if (node.getKind() == NodeKind.INVOCATION && node.getDeterminedType().getKind() == TypeKind.FUTURE) {
+            BLangInvocation.BLangActionInvocation invocationNode = (BLangInvocation.BLangActionInvocation) node;
+
+            if (invocationNode.isAsync()
+                    && PositionUtil.withinBlock(range.startLine(), invocationNode.getName().getPosition())) {
+
+                return ((BFutureType) node.getDeterminedType()).getConstraint();
+            }
+        }
```

Comment:
This will crash for something like the following:
```ballerina
public function test() {
    future<int> futureResult = <1>foo()<2>;
}

function foo() returns future<int> {
    return start bar();
}

function bar() returns int {
    return 10;
}
```